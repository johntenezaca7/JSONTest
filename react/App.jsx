import React from "react";
import { Route } from "react-router-dom";
import { Provider } from "react-redux";
import store from "./store";

import Landing from "./Landing";
import Test from "./Assessment";
import Done from "./Done";

const App = () => (
  <Provider store={store}>
    <div className="app">
      <Route exact path="/" component={props => <Landing />} />
      <Route exact path="/assessment" component={props => <Test />} />
      <Route exact path="/done" component={props => <Done />} />
    </div>
  </Provider>
);

export default App;
