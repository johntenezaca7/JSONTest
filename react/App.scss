$primary-color1: rgb(255, 255, 255);
$primary-font: 'Gugi', cursive;
$second-font :'Roboto', sans-serif;

* {
    box-sizing: border-box
}
body {
    margin: 0%;
    font-family: $second-font;
    background-color: $primary-color1;
    color: rgb(32, 32, 32);
    overflow:hidden;
  }
  img {
    cursor: pointer;
  }
  
  a {
    color: black;
    text-decoration: none;
    cursor: pointer;
  }
  input {
      padding: 0px;
  }

  video {
      position: absolute;
      top: 0;
      left: 0;
      width: 100%;
      overflow: hidden;
      
  }
  .video-overlay {
      height: 100%;
      width: 100%;
      position: absolute;
      top:0;
      left: 0;
      background-color: rgb(53, 87, 126);
      z-index:1;
      opacity: 0.85;   
      overflow: hidden;
  }

  .landing-video{
      display: gird;
      grid-template-columns: auto;
      justify-content: center;
  }

  .landing {
    background-color: inherit;
    position: absolute;
    top:25%;
    left:25%;
    right:25%;
    margin: auto;
    display: grid;
    grid-template-columns: auto;
    z-index: 1;
    color: white;
    align-self: center;

    h2 {
        font-family: $primary-font; 
        font-size: 50px;
        margin: 0%;
        justify-self: center;
    }

    .GetName {
        text-align: center;
        margin-bottom: 2%;
        font-family: $primary-font;
        justify-self: center
    }

    div {
    justify-self: center;
    }
}
  
  .inside-landing input {
    height: 40px;
    font-size: 20px;
    text-align: center;
  }
  .landing-btn {
    margin-top: 5%;
    background-color: rgb(255, 255, 255); 
    border: none;
    width: 100%;
    color: white;
    padding: 0px;
    height: 40px;
    text-align: center;
    text-decoration: none;
    display: inline-block;
    font-size: 16px;
    -webkit-transition-duration: 0.4s; /* Safari */
    transition-duration: 0.4s;
    cursor: pointer;
    justify-self: center
  }
  
  .landing-btn {
    background-color: white;
    color: black;
  }
  
  .landing-btn:hover {
    background-color: rgb(35, 55, 78);
    color: white;
  }
  
  .test-display h2 {
    text-align: center;
    margin-bottom: 5%;
  }
  
  .test-display {
    display: grid;
    grid-template-columns: auto;
    grid-template-rows: 1fr 4fr
  }
  
  .test-box {
    height: 500px;
    width: 900px;
    border: 1px solid rgb(250, 250, 250);
    background-color: white;
    border-radius: 2%;
    box-shadow: -2px 3px 5px rgba(77, 78, 79, 0.2);
    justify-self: center;
  }
  
  .test-details{
    text-align: center;
  }
  .test-details h4 {
    font-size: 25px;
    margin: 1%;
  }
  .test-details p {
    font-size: 16px;
    line-height: 1.5;
  }
  .container {
    padding: 5%;
  }
  .back-n-count {
    display: grid;
    grid-template-columns: 1fr 1fr;
  }
  
  .back-n-count p {
    justify-self: end;
  }
  
  .answers {
    display: flex;
    flex-flow: row wrap;
    justify-content: center
  }
  
  .answer-box {
    border: 1px solid grey;
    width: 180px;
    height: 100px;
    border-radius: 15px;  
    box-shadow: -2px 3px 5px rgba(77, 78, 79, 0.2);
    margin: 1%;
    cursor: pointer;
    display: flex;
    flex-flow: row wrap;
    justify-content: center
  }
  
  .answer-box p {
    align-self: center;
    padding-left: 5px;
  
  }