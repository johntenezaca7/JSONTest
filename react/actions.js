export const SET_SEARCH_TERM = "SET_SEARCH_TERM";
export const INCREMENT = "INCREMENT";
export const DECREMENT = "DECREMENT";
export const RESTART = "RESTART";
