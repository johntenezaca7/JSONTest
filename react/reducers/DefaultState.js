export const DEFAULT_STATE = {
  searchTerm: "",
  score: 0,
  questions: [
    {
      title: "1. Inside which HTML element do we put the JavaScript?",
      description: "",
      choices: [
        ["<script>", 5],
        ["<javascript>", 1],
        ["<scripting>", 1],
        ["<js>", 1]
      ],
      done: false
    },
    {
      title: "2. What is the correct way to write a JavaScript array?",
      description: "",
      choices: [
        ["var colors = (1:'red', 2:'green', 3:'blue')", 1],
        ["var colors = 1 = ('red'), 2 = ('green'), 3 = ('blue')", 1],
        ["var colors = 'red', 'green', 'blue'", 1],
        ["var colors = ['red', 'green', 'blue']", 5]
      ],
      done: false
    },
    {
      title: "3. How do you find the number with the highest value of x and y?",
      description: "",
      choices: [
        ["Math.ceil(x, y)", 1],
        ["ceil(x, y)", 1],
        ["top(x, y)", 1],
        ["Math.max(x, y)", 5]
      ],
      done: false
    },
    {
      title: "4. What is React?",
      description: "React is a front end JavaScript library developed by Facebook in 2011. It follows the component based approach which helps in building reusable UI components. It is used for developing complex and interactive web and mobile UI. Even though, it was open-sourced only in 2015, it has one of the largest communities supporting it.",
      choices: [["Yes", 5], ["No", 0]],
      done: true
    }
  ]
};
