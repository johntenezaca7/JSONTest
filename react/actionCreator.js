import { SET_SEARCH_TERM, INCREMENT, DECREMENT, RESTART } from "./actions";

export function setSearchTerm(searchTerm) {
  return { type: SET_SEARCH_TERM, payload: searchTerm };
}

export function incrementScore(score) {
  return { type: INCREMENT, payload: score };
}

export function decrementScore(score) {
  return { type: DECREMENT, payload: score };
}

export function restart() {
  return { type: RESTART, payload: 0 };
}
