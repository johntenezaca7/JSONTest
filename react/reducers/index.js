import { combineReducers } from "redux";
import { SET_SEARCH_TERM, INCREMENT, DECREMENT, RESTART } from "../actions";
import { DEFAULT_STATE } from "./DefaultState";

const setSearchTerm = (state, action) => {
  return Object.assign({}, state, { searchTerm: action.payload });
};

const incScore = (state, action) => {
  return Object.assign({}, state, { score: state.score + action.payload });
};

const decScore = (state, action) => {
  return Object.assign({}, state, { score: state.score - action.payload });
};

const restart = (state, action) => {
  return Object.assign({}, state, { score: action.payload });
};

const rootReducer = (state = DEFAULT_STATE, action) => {
  switch (action.type) {
    case SET_SEARCH_TERM:
      return setSearchTerm(state, action);
    case INCREMENT:
      return incScore(state, action);
    case DECREMENT:
      return decScore(state, action);
    case RESTART:
      return restart(state, action);
    default:
      return state;
  }
};

export default rootReducer;
