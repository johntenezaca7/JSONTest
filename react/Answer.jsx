import React from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { Link } from "react-router-dom";
import { incrementScore } from "./actionCreator";

const Answer = props => {
  const { choice, incScore, onClick, current } = props;

  const updateScore = () => {
    let score = choice[1];
    incScore(score);
  };

  const fireAll = () => {
    updateScore();
    onClick();
  };
  let isDone;

  if (current.done) {
    isDone = (
      <div className="answer-box" onClick={updateScore}>
        <Link to="/done">
          <p>{choice[0]}</p>
        </Link>
      </div>
    );
  } else {
    isDone = (
      <div className="answer-box" onClick={fireAll}>
        <p>{choice[0]}</p>
      </div>
    );
  }
  return isDone;
};

Answer.propTypes = {
  incScore: PropTypes.func,
  current: PropTypes.object,
  choice: PropTypes.array,
  onClick: PropTypes.func
};

const mapDispatchtoProps = dispatch => ({
  incScore(score) {
    dispatch(incrementScore(score));
  }
});

export const Unwrapped = Answer;

export default connect(null, mapDispatchtoProps)(Answer);
