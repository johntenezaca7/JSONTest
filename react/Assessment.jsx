import React, { Component } from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { decrementScore } from "./actionCreator";
import Answer from "./Answer";

class Assessment extends Component {
  constructor(props) {
    super(props);
    this.state = {
      index: 0,
      testNum: 1,
      current: this.props.questions[0],
      savedScore: ""
    };

    this._onRightClick = this._onRightClick.bind(this);
    this._onLeftClick = this._onLeftClick.bind(this);
    this._run = this._run.bind(this);
  }

  _run(direction) {
    let nextValue, questions;

    if (direction === "right") {
      if (this.state.index <= 4) {
        this.state.index += 1;
        this.state.testNum += 1;

        nextValue = this.state.index;
        questions = this.props.questions;

        this.setState((prev, next) => {
          let savedVal = prev.savedScore ? prev.savedScore : next.currentScore;
          return {
            current: questions[nextValue],
            savedScore: savedVal
          };
        });
      }
    } else if (direction === "left") {
      if (this.state.index !== 0) {
        this.state.index -= 1;
        this.state.testNum -= 1;

        nextValue = this.state.index;
        questions = this.props.questions;

        this.setState((prev, next) => {
          return {
            current: questions[nextValue],
            savedScore: prev.savedScore
          };
        });
      }
    }
  }

  _onRightClick() {
    this._run("right");
  }

  _onLeftClick() {
    this._run("left");
    this.props.decScore(this.state.savedScore);
  }

  render() {
    const { _onLeftClick, _onRightClick } = this;
    const { testNum, current } = this.state;
    const { name } = this.props;

    let checkName;
    let checkNav;

    if (name) {
      checkName = name;
    } else {
      checkName = "Anonymous";
    }

    if (testNum === 1) {
      checkNav = <div />;
    } else {
      checkNav = (
        <img
          onClick={_onLeftClick}
          src="https://image.ibb.co/jyA5n7/prev.png"
          alt="Back"
        />
      );
    }
    return (
      <div className="test-display">
        <h2>Welcome {checkName}!</h2>
        <div className="test-box">
          <div className="container">
            <div className="back-n-count">
              {checkNav}
              <p>Step {testNum} of 4</p>
            </div>
            <div className="test-details">
              <h4>{current.title}</h4>
              <p>{current.description}</p>
            </div>
            <div className="answers">
              {current.choices.map((ele, i) => {
                return (
                  <Answer
                    choice={ele}
                    key={i}
                    onClick={_onRightClick}
                    current={current}
                  />
                );
              })}
            </div>
          </div>
        </div>
      </div>
    );
  }
}

Assessment.propTypes = {
  currentScore: PropTypes.number,
  decScore: PropTypes.func,
  questions: PropTypes.array,
  name: PropTypes.string
};

const mapStateToProps = state => ({
  questions: state.questions,
  name: state.searchTerm,
  currentScore: state.score
});

const mapDispatchtoProps = dispatch => ({
  decScore(score) {
    dispatch(decrementScore(score));
  }
});

export const Unwrapped = Assessment;

export default connect(mapStateToProps, mapDispatchtoProps)(Assessment);
