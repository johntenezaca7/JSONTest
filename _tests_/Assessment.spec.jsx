import React from "react";
import jest from "jest";
import { shallow, render } from "enzyme";
import { DEFAULT_STATE } from "../react/reducers/DefaultState";
import { Unwrapped as UnwrappedAssessment } from "../react/Assessment";
import Answer from "../react/Answer";

test("Assessment Component renders correctly with one question", () => {
  const component = shallow(
    <UnwrappedAssessment
      currentScore={20}
      decScore={f => f}
      questions={[
        {
          title: "1. Inside which HTML element do we put the JavaScript?",
          description: "",
          choices: [
            ["<script>", 5],
            ["<javascript>", 1],
            ["<scripting>", 1],
            ["<js>", 1]
          ],
          done: false
        }
      ]}
      name={"test"}
    />
  );
  expect(component).toMatchSnapshot();
});

test("Assessment Component renders the right about of answer choices.", () => {
  const component = shallow(
    <UnwrappedAssessment
      currentScore={20}
      decScore={f => f}
      questions={DEFAULT_STATE.questions}
      name={"test"}
    />
  );
  expect(component.find(Answer).length).toEqual(
    DEFAULT_STATE.questions[0].choices.length
  );
});
