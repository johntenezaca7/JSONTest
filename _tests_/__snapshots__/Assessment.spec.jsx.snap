// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Assessment Component renders correctly with one question 1`] = `
<div
  className="test-display"
>
  <h2>
    Welcome 
    test
    !
  </h2>
  <div
    className="test-box"
  >
    <div
      className="container"
    >
      <div
        className="back-n-count"
      >
        <div />
        <p>
          Step 
          1
           of 4
        </p>
      </div>
      <div
        className="test-details"
      >
        <h4>
          1. Inside which HTML element do we put the JavaScript?
        </h4>
        <p />
      </div>
      <div
        className="answers"
      >
        <Connect(Answer)
          choice={
            Array [
              "<script>",
              5,
            ]
          }
          current={
            Object {
              "choices": Array [
                Array [
                  "<script>",
                  5,
                ],
                Array [
                  "<javascript>",
                  1,
                ],
                Array [
                  "<scripting>",
                  1,
                ],
                Array [
                  "<js>",
                  1,
                ],
              ],
              "description": "",
              "done": false,
              "title": "1. Inside which HTML element do we put the JavaScript?",
            }
          }
          onClick={[Function]}
        />
        <Connect(Answer)
          choice={
            Array [
              "<javascript>",
              1,
            ]
          }
          current={
            Object {
              "choices": Array [
                Array [
                  "<script>",
                  5,
                ],
                Array [
                  "<javascript>",
                  1,
                ],
                Array [
                  "<scripting>",
                  1,
                ],
                Array [
                  "<js>",
                  1,
                ],
              ],
              "description": "",
              "done": false,
              "title": "1. Inside which HTML element do we put the JavaScript?",
            }
          }
          onClick={[Function]}
        />
        <Connect(Answer)
          choice={
            Array [
              "<scripting>",
              1,
            ]
          }
          current={
            Object {
              "choices": Array [
                Array [
                  "<script>",
                  5,
                ],
                Array [
                  "<javascript>",
                  1,
                ],
                Array [
                  "<scripting>",
                  1,
                ],
                Array [
                  "<js>",
                  1,
                ],
              ],
              "description": "",
              "done": false,
              "title": "1. Inside which HTML element do we put the JavaScript?",
            }
          }
          onClick={[Function]}
        />
        <Connect(Answer)
          choice={
            Array [
              "<js>",
              1,
            ]
          }
          current={
            Object {
              "choices": Array [
                Array [
                  "<script>",
                  5,
                ],
                Array [
                  "<javascript>",
                  1,
                ],
                Array [
                  "<scripting>",
                  1,
                ],
                Array [
                  "<js>",
                  1,
                ],
              ],
              "description": "",
              "done": false,
              "title": "1. Inside which HTML element do we put the JavaScript?",
            }
          }
          onClick={[Function]}
        />
      </div>
    </div>
  </div>
</div>
`;
